### PR TITLE
Explain EventWriter limits concurrency

### PR DESCRIPTION
**Pull request by stepancheg**

### CODE REVIEW

File: crates/bevy_ecs/src/event.rs [hunk 1]
```diff
@@ -484,7 +484,15 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
 ///
-/// # Limitations
+/// # Concurrently
```
Comment (mockersf) [inline crates/bevy_ecs/src/event.rs:487]:
```suggestion
/// # Concurrency
```

File: crates/bevy_ecs/src/event.rs [hunk 2]
```diff
@@ -484,7 +484,15 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
...
 /// ```
 ///
-/// # Limitations
+/// # Concurrency
+///
+/// `EventWriter` param has [`ResMut<Events<T>>`](Events) inside. So two systems declaring `EventWriter<T>` params
+/// for the same `<T>` won't be executed concurrently.
+///
+/// This does not apply to [`EventReader<T>`](EventReader) though: reader systems are executed concurrently
+/// (but not with `EventWriter<T>` systems).
```
Comment (Nilirad) [inline crates/bevy_ecs/src/event.rs:493]:
```suggestion
```
This line repeats what was stated a couple of lines before.

**Review by Nilirad — CHANGES_REQUESTED** [review]

**Review by Nilirad — APPROVED** [review]

File: crates/bevy_ecs/src/event.rs [hunk 3]
```diff
@@ -484,7 +489,12 @@ impl<'w, 's, E: Event> EventReader<'w, 's, E> {
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
 ///
-/// # Limitations
+/// # Concurrency
+///
+/// `EventWriter` param has [`ResMut<Events<T>>`](Events) inside. So two systems declaring `EventWriter<T>` params
+/// for the same `<T>` won't be executed concurrently.
```
Comment (james7132) [inline crates/bevy_ecs/src/event.rs:495]:
```suggestion
/// for the same event type won't be executed concurrently.
```

File: crates/bevy_ecs/src/event.rs [hunk 4]
```diff
@@ -407,6 +407,11 @@ impl<E: Event> DerefMut for EventSequence<E> {
 }
 
 /// Reads events of type `T` in order and tracks which events have already been read.
+///
+/// # Concurrency
+///
+/// Unlike [`EventWriter<T>`], systems with `EventReader<T>` param can be executed concurrently
+/// (but not concurrently with `EventWriter<T>` systems for the same `<T>`).
```
Comment (james7132) [inline crates/bevy_ecs/src/event.rs:414]:
```suggestion
/// (but not concurrently with `EventWriter<T>` systems for the same event type).
```

**Review by james7132 — APPROVED** [review]

**Pull request merged**